topdown/sets.go

```diff
@@ -23,6 +25,48 @@ func builtinSetDiff(a, b ast.Value) (ast.Value, error) {
 	return s1.Diff(s2), nil
 }
 
+// builtinSetIntersection returns the intersection of the given input sets
+func builtinSetIntersection(a ast.Value) (ast.Value, error) {
+	switch a := a.(type) {
+	case ast.Set:
+		result := a.Keys()[0].Value.(ast.Set)
```

Comment:
If the expression is `intersection(set())` this will panic. The output in this case should be an empty set.